phoenix-core/src/main/java/org/apache/phoenix/schema/types/S3BasedLobStore.java

```diff
@@ -0,0 +1,69 @@
+package org.apache.phoenix.schema.types;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.S3Object;
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.RandomStringUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class S3BasedLobStore implements LobStore {
+
+    private static final String ROOT_DIRECTORY = "/tmp/";
+    private AmazonS3 s3client;
+    private static String BUCKET_NAME = "phoenixgiants-bucket";
+    private static S3BasedLobStore INSTANCE = null;
+
+    public synchronized static S3BasedLobStore getInstance() {
+        if(INSTANCE == null){
+            INSTANCE = new S3BasedLobStore();
+        }
+        return INSTANCE;
+    }
+
+    public S3BasedLobStore() {
+        BasicAWSCredentials credentials = new BasicAWSCredentials("AKIAIRFVDBF3ZVSZDP4Q", "PPLkr8QCRWlPWBYzJQeR2NkJGImg/NVupAD4tGjf");
+        s3client = AmazonS3ClientBuilder.standard()
+                        .withCredentials(new AWSStaticCredentialsProvider(credentials))
+                        .withRegion(Regions.US_WEST_2).build();
+    }
+
+    @VisibleForTesting
+    String getPathFromLocator(String lobLocator) {
+        return ROOT_DIRECTORY + lobLocator;
+    }
+
+    @Override
+    public String putLob(InputStream lobStream) throws LobStoreException {
+        String lobLocator = RandomStringUtils.randomAlphanumeric(15); //For now use lobLocator
+        File tempStore = new File(getPathFromLocator(lobLocator));
+        try {
+            FileUtils.copyInputStreamToFile(lobStream, tempStore);
```

Comment:
Do we need the tempfile? I see some syntax around `PutObjectRequest` which can maybe directly put from an inputStream.

Reply:
+1 for no temporary file if possible

Reply:
Yeah I had come across PutObjectRequest which accepts InputStream directly but it also requires ObjectMetaData param(Which can also be null or new ObjectMetaData ). I wasn't very sure about it so I took easier approach. I can change it to PutObjectRequest after we discuss more about ObjectMetaData.